src/V3ProtectLib.cpp

```diff
@@ -426,6 +426,15 @@ class ProtectVisitor final : public VNVisitor {
 
     void visit(AstVar* nodep) override {
         if (!nodep->isIO()) return;
+
+        if (VN_IS(nodep->dtypep()->skipRefp(), NodeUOrStructDType)) {
```

Comment:
This will work for unpacked structs only, probably should check that (or improve it to work for both - which will be a lot more work)

Reply:
Sorry, do you mean work for **packed** structs only? The included test has a packed struct. I didn't even realize verilator supported unpacked structs.

This does bring up the point that this doesn't support arrays of packed structs. Do you have suggestions on how to do that?

Reply:
Ah, yes I meant this works only for packed, not for unpacked.  For complicated data types the right way is to recurse on the data type, building up a new data type for the related call.  An example of this is AstNodeDType::cTypeRecurse